File: 07_ServerlessFargatePipeline/typescript/lib/serverless_fargate_pipeline-stack.ts
import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import * as codecommit from 'aws-cdk-lib/aws-codecommit';
import {CodeBuildStep, CodePipeline, CodePipelineSource} from "aws-cdk-lib/pipelines";
import {ServerlessStage} from './serverless-stage';
import {FargateStage} from './fargate-stage';

export class ServerlessFargatePipelineStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    const repo = new codecommit.Repository(this, 'cdkPipelinesRepo', {
      repositoryName: "cdkPipelinesRepo"
    });

    const pipeline = new CodePipeline(this, 'Pipeline', {
      pipelineName: 'WorkshopPipeline',
      synth: new CodeBuildStep('SynthStep', {
              input: CodePipelineSource.codeCommit(repo, 'master'),
              installCommands: [
                  'npm install -g aws-cdk'
              ],
              commands: [
                  'npm ci',
                  'npm run build',
                  'npx cdk synth'
              ]
          }
      )
    });

    const serverless = new ServerlessStage(this, 'ServerlessStage');
    const serverless_stage = pipeline.addStage(serverless);

    const fargate = new FargateStage(this, 'FargateStage');
    const fargate_stage = pipeline.addStage(fargate);
  }
}

File: 08_ProjenServerlessFargatePipeline/python/ServerlessFargatePipeline/serverless_fargate_pipeline_stack.py
from constructs import Construct
from aws_cdk import (
    Stack,
    aws_codecommit as codecommit,
    pipelines as pipelines,
)

from constructs import Construct
from .serverless_stage import ServerlessStage
from .fargate_stage import FargateStage

class ServerlessFargatePipelineStack(Stack):

    def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
        super().__init__(scope, construct_id, **kwargs)
        
        repo = codecommit.Repository(
            self, 'AymanRepoCDK',
            repository_name= "AymanRepoCDK"
        )

        pipeline = pipelines.CodePipeline(
            self,
            "AymanPipelineCDK",
            synth=pipelines.ShellStep(
                "Synth",
                input=pipelines.CodePipelineSource.code_commit(repo, "master"),
                commands=[
                    "npm install -g aws-cdk",  # Installs the cdk cli on Codebuild
                    "pip install -r requirements.txt",  # Instructs Codebuild to install required packages
                    "npx cdk synth",
                ]
            ),
        )

        serverless = ServerlessStage(self, "ServerlessStage")
        serverless_stage = pipeline.add_stage(serverless)

        fargate = FargateStage(self, "FargateStage")
        fargate_stage = pipeline.add_stage(fargate)


